### PR TITLE
Helm v3

### LINKED ISSUE #1502: Upgrade to Helm 3
**Problem**
Our charts are still using the Helm 2 chart format. Helm 2 is deprecated. Luckily we use Helm 3 to actually deploy.

**Solution**
- [Migrate](https://helm.sh/docs/topics/v2_v3_migration/) chart config to helm 3
- Migrate Marketplace to v3
- Test migrated chart

**Alternatives**

**Additional Context**

### PR DESCRIPTION
**Pull request by Nana-EC**
**Detailed description**:
Update chart config and marketplace to Helm 3

- Update `Chart.yaml` files to `apiVersion: v2`
- Remove `requirements` files and move dependencies to `Chart.yaml`
- Update multimode pom replace logic to handle recommended quotation of v3 `appVersion` label

**Which issue(s) this PR fixes**:
Fixes #1502 

**Special notes for your reviewer**:
GCP marketplace doesn't need updating as marketplace deployer has been using helm v3 deployer since Jun 2020 - https://github.com/GoogleCloudPlatform/marketplace-k8s-app-tools/blob/master/docs/building-deployer-helm.md#build-your-deployer-container

Verified deployer docker image creation works with v3 flow.
Verified `mp-dev` install and run of chart was successful

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: charts/hedera-mirror/Chart.lock [hunk 1]
```diff
@@ -20,5 +20,5 @@ dependencies:
 - name: timescaledb-multinode
   repository: https://raw.githubusercontent.com/timescale/timescaledb-kubernetes/master/charts/repo/
   version: 0.8.0
-digest: sha256:2bb1e5793b8d3af4509ad3e1744b0a411d00e53d68f423f78bdbbd9ef37625cf
-generated: "2021-03-23T20:42:17.054694-05:00"
+digest: sha256:6d0fca0e132d37183e85fbb7fe61f6cf92b3c9728baaf06edf779750d72d7a7f
+generated: "2021-03-24T11:55:13.398028-05:00"
```
Comment (xin-hedera) [inline charts/hedera-mirror/Chart.lock:24]: missing `Chart.lock` for other charts

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: charts/hedera-mirror-common/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,24 @@
-apiVersion: v1
-appVersion: 0.30.0-rc1
+apiVersion: v2
+appVersion: "0.30.0-rc1"
```
[the remark this answers is not in the document]
Comment (xin-hedera) [reply charts/hedera-mirror-common/Chart.yaml:2]: Just saw from helm v3.5.0, `helm create` wraps the default `appVersion` field in quotes.

[the remark this answers is not in the document]
Comment (Nana-EC) [reply charts/hedera-mirror-common/Chart.yaml:2]: So it's not necessary but recommended in v3, apparently "Leaving it unquoted can lead to parsing issues in some cases." - See https://helm.sh/docs/topics/charts/#the-appversion-field

File: charts/hedera-mirror/Chart.yaml [hunk 3]
```diff
@@ -1,8 +1,54 @@
...
+    condition: timescaledb.enabled
+    name: timescaledb-multinode
+    repository: https://raw.githubusercontent.com/timescale/timescaledb-kubernetes/master/charts/repo/
+    version: ^0.8.0
 description: Hedera Mirror Node mirrors data from Hedera nodes and serves it via an API
 home: https://github.com/hashgraph/hedera-mirror-node
 icon: https://camo.githubusercontent.com/cca6b767847bb8ca5c7059481ba13a5fc81c5938/68747470733a2f2f7777772e6865646572612e636f6d2f6c6f676f2d6361706974616c2d686261722d776f72646d61726b2e6a7067
+keywords:
+  - blockchain
+  - dtl
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply charts/hedera-mirror/Chart.yaml:48]: Will update

File: charts/hedera-mirror/Chart.lock (hunk 1 above)
Comment (Nana-EC) [reply charts/hedera-mirror/Chart.lock:24]: Thanks will add

File: charts/hedera-mirror/Chart.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nana-EC) [reply charts/hedera-mirror/Chart.yaml:48]: Actually all of the tags seem to be lowercase by design. Unless I'm incorrect @steven-sheehy

[the remark this answers is not in the document]
Comment (xin-hedera) [reply charts/hedera-mirror/Chart.yaml:48]: in that case, it should be `dlt`, right?

File: charts/hedera-mirror/Chart.lock (hunk 1 above)
Comment (Nana-EC) [reply charts/hedera-mirror/Chart.lock:24]: Added

File: charts/hedera-mirror/Chart.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nana-EC) [reply charts/hedera-mirror/Chart.yaml:48]: Haha yes you're right. Missed that

**Review by steven-sheehy — APPROVED** [review]
LGTM

File: charts/hedera-mirror/Chart.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nana-EC) [reply charts/hedera-mirror/Chart.yaml]: Fixed

**Review by steven-sheehy — APPROVED** [review]

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by ijungmann — APPROVED** [review]

**Pull request merged**